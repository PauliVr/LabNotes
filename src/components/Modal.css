.modal {
  position: absolute;
  inset: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal__container {
  background-color: #ffffff;
  border-radius: 35px;
  display: flex;
  flex-direction: column;
  align-self: center;
  width: 450px;
  height: 550px;
}

.modal__img {
  width: 200px;
  align-self: center;
}

.btn__close {
  background-color: transparent;
  border-style: none;
  align-self: flex-end;
  padding: 15px 25px;
}

.btn__close img {
  width: 35px;
  height: 35px;
}

.modal__title {
  color: #d880a6;
  padding: 0 25px;
  margin: 25px;
  font-size: 24px;
}

.modal__text {
  color: #595757;
  margin: 10px;
}

.modal__btns {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 25px 0;
}

.modal__btnCancel,
.delete {
  background-color: #d880a6;
  color: #ffffff;
  font-family: 'Montserrat', sans-serif;
  font-size: 20px;
  border-style: none;
  padding: 10px;
  border-radius: 20px;
}

.modal__btnCancel {
  box-shadow: 1px 1px 1px 1px #7e3793;
}

.delete {
  box-shadow: 1px 1px 1px 1px #d63197;
}

.delete:hover {
  background-color: #d63197;
}

.modal__btnCancel:hover {
  background-color: #7e3793;
}
