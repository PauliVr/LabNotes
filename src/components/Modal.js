import deleteNote from '../firebase/firebase';
import '../globalStyles.css';
import './Modal.css';

export default function Modal(props) {
  async function deletedNote(id) {
    if (id) {
      await deleteNote(id);
      props.delete(false, '');
    }
    console.log('se elimino la nota con id: ' + id);
    // setTimeout(() => {
    //   props.delete(false, '');
    // }, 3000);
  }

  return (
    <section className='modal'>
      <div className='modal__container'>
        <button
          type='button'
          className='btn__close'
          onClick={() => {
            props.delete(false, '');
          }}
        >
          close
          <img className='modal__img' src='./assets/close.svg' alt='IconClose' />
        </button>
        <img className='modal__img' src='./assets/GatoAlert.svg' alt='AstroCat' />
        <h3 className='modal__title'>¿Estas segur@ que quieres eliminar la nota?</h3>
        <p className='modal__text'>* Al eliminar la nota no podrás recuperarla * </p>
        <div className='modal__btns'>
          <button
            className='modal__btnCancel'
            onClick={() => {
              props.delete(false, '');
            }}
          >
            Cancelar
          </button>
          <button
            className='delete'
            onClick={() => {
              deletedNote(props.id);
            }}
          >
            Eliminar Nota
          </button>
        </div>
      </div>
    </section>
  );
}
