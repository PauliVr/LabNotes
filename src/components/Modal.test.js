/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render } from '@testing-library/react';
import { prettyDOM } from '@testing-library/react';
import Modal from './Modal';

test('Renders modal component', () => {
  const view = render(<Modal />);
  view.getAllByAltText('AstroCat');
  view.getByText('¿Estas segur@ que quieres eliminar la nota?');
  expect(view.container).toHaveTextContent('* Al eliminar la nota no podrás recuperarla * ');
  // eslint-disable-next-line testing-library/no-debugging-utils
  view.debug();
});

test('Clicking the button calls event handler once', () => {
  const del = false;
  const mockHandler = jest.fn();
  const view = render(<Modal onClick={mockHandler} />);
  expect(view).toMatchSnapshot();
  const button = view.getByText('Eliminar Nota');
  expect(mockHandler.mock.calls).toHaveLength(0);
});
