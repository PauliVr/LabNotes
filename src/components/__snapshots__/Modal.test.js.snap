// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking the button calls event handler once 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="modal"
      >
        <div
          class="modal__container"
        >
          <button
            class="btn__close"
            type="button"
          >
            close
            <img
              alt="IconClose"
              class="modal__img"
              src="./assets/close.svg"
            />
          </button>
          <img
            alt="AstroCat"
            class="modal__img"
            src="./assets/GatoAlert.svg"
          />
          <h3
            class="modal__title"
          >
            ¿Estas segur@ que quieres eliminar la nota?
          </h3>
          <p
            class="modal__text"
          >
            * Al eliminar la nota no podrás recuperarla * 
          </p>
          <div
            class="modal__btns"
          >
            <button
              class="modal__btnCancel"
            >
              Cancelar
            </button>
            <button
              class="delete"
            >
              Eliminar Nota
            </button>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="modal"
    >
      <div
        class="modal__container"
      >
        <button
          class="btn__close"
          type="button"
        >
          close
          <img
            alt="IconClose"
            class="modal__img"
            src="./assets/close.svg"
          />
        </button>
        <img
          alt="AstroCat"
          class="modal__img"
          src="./assets/GatoAlert.svg"
        />
        <h3
          class="modal__title"
        >
          ¿Estas segur@ que quieres eliminar la nota?
        </h3>
        <p
          class="modal__text"
        >
          * Al eliminar la nota no podrás recuperarla * 
        </p>
        <div
          class="modal__btns"
        >
          <button
            class="modal__btnCancel"
          >
            Cancelar
          </button>
          <button
            class="delete"
          >
            Eliminar Nota
          </button>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
